src/Lykke.Job.EthereumCore/Job/MonitoringCoinTransactionJob.cs

```diff
@@ -211,11 +225,43 @@ await _eventTraceRepository.InsertAsync(new EventTrace()
                     default: break;
                 }
 
+                #region FailedCashout
+
                 if (coinEvent.CoinEventType == CoinEventType.CashoutCompleted && !success)
                 {
                     coinEvent.CoinEventType = CoinEventType.CashoutFailed;
+                    await _coinEventService.InsertAsync(coinEvent);
+                    SendMessageToTheQueueEnd(context, transaction, 200, "Put Failed cashout in the end of the queue");
+
+                    return false;
                 }
 
+                if (coinEvent.CoinEventType == CoinEventType.CashoutFailed && !success)
+                {
+                    var historycal = await _pendingOperationService.GetHistoricalAsync(operationId);
+
+                    if (historycal != null && historycal.Count() != 0)
```

Comment:
check for .Coun() != 0 is unnecessary, foreach will stop execution is there are no records